### PR TITLE
Fix json files in index.js and auto-deassign number

### PR DESCRIPTION
**Pull request by NubbyTM**
I did it right this time and did it once every month with a notice and everything austin asked. I left a little thing that would put it in the bot-log channel but commented it out because it would get quite spammy.

nubbytm

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -26,6 +27,37 @@ server.listen(port, ipaddress, () => {
...
+	numbers.splice(numberIndex, 1);
+}
+
+schedule.scheduleJob({date: 1, hour: 0, minute: 0, second: 0}, function(){
+	const now = new Date();
+	for (var i in numbers){
+		const number = numbers[i];
+		if (number.year <= now.getFullYear() || number.month <= now.getMonth()){
+			if (number.month == now.getMonth() || (number.month == 12 && now.getMonth() == 0)){
+				// send a notice to the user.
```
Comment (KingDGrizzle) [inline index.js]: This is not how you get channels ever. You use get(ID), where id is number.channel in this case

File: index.js [hunk 2]
```diff
@@ -26,6 +27,37 @@ server.listen(port, ipaddress, () => {
...
+				var channel = bot.channels.get("id", number.channel);
+				if (channel != null){ // if the channel is null we will remove them because that means deleted. :(
+					var message = "Your number is expired! Pay your monthly fee by typing `>dial *233`!";
+					channel.send(message);
+					break;
+				}
+			}
+			removeNumber(i);
+			// Uncomment if I should log it. I don't think it would be a good idea because it happens every month, so spam. - nubbytm
```
Comment (KingDGrizzle) [inline index.js]: Same as above

**Comment by NubbyTM** [comment]
@KingDGrizzle fixed. i kind-of thought it worked more like a collection/map or whatever. thanks for the quick catch™

File: index.js [hunk 3]
```diff
@@ -26,6 +27,37 @@ server.listen(port, ipaddress, () => {
...
+	const now = new Date();
+	for (var i in numbers){
+		const number = numbers[i];
+		if (number.year <= now.getFullYear() || number.month <= now.getMonth()){
+			if (number.month == now.getMonth() || (number.month == 12 && now.getMonth() == 0)){
+				// send a notice to the user.
+				var channel = bot.channels.get(number.channel);
+				if (channel != null){ // if the channel is null we will remove them because that means deleted. :(
+					var message = "Your number is expired! Pay your monthly fee by typing `>dial *233`!";
+					channel.send(message);
```
Comment (austinhuang0131) [inline index.js]: u can't send msg if channel is null

also it's undefined not null

File: index.js [hunk 4]
```diff
@@ -26,6 +27,37 @@ server.listen(port, ipaddress, () => {
 
 const mailbox_storage = JSON.parse(fs.readFileSync("./mailbox.json", "utf8"));
 
+function updateNumbers(){
+	fs.writeFileSync("./json/numbers.json", JSON.stringify(numbers), "utf8");
+	numbers = JSON.stringify(numbers);
```
Comment (austinhuang0131) [inline index.js]: remove this line

**Comment by austinhuang0131** [comment]
i meant that u auto-deassign if it has been expired for a month

also check comments

File: index.js (hunk 3 above)
Comment (NubbyTM) [reply index.js]: wait wut ur telling me undefined isn't null. eh i'll just do !channel. i have been doing too much java.

**Pull request merged**